Regular-Question-Answers/symmetric-tree.cpp

```diff
@@ -0,0 +1,70 @@
+// Author: Ceren Yaşar
+// Question Link: https://leetcode.com/problems/symmetric-tree/
+
+// Time Complexity: O(N)
+// Memory Complexity: O(N)
+
+// This solution uses breadth first search since we start comparisons from the closest child.
+
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
+ * };
+ */
+
+class Solution {
+public:
+    bool isSymmetric(TreeNode* root) {
+        
+        // This queue keeps pairs of tree nodes that are supposed to have equal values
+        queue<pair<TreeNode*, TreeNode*>> nextToCheck;
+        
+        // If the root node stands alone, it is symmetric
+        if (!root->left && !root->right)
+            return true;
+        // If the root node has only one child, it is asymmetric
+        else if (!root->left || !root->right)
+            return false;
+        
+        // First pair that is supposed to have equal values is direct children of the root
+        nextToCheck.push({root->left,root->right});
+        
+        TreeNode *a, *b;
+
+        while (!nextToCheck.empty()) {
+
+            a = nextToCheck.front().first;
+            b = nextToCheck.front().second;
+            nextToCheck.pop();
+            
+            // If a and b have equal values, we can look for next nodes to check which would be
+            // direct children of a's and b's.
```

Comment:
It should be either `// direct children of a and b` or `// a's and b's direct children`.